integration-tests/src/test/java/com/example/NestedDependencyInterfaceTest.java

```diff
@@ -0,0 +1,25 @@
+package com.example;
+
+import dagger.Component;
+
+@Component(dependencies = NestedDependencyInterfaceTest.First.class)
+public interface NestedDependencyInterfaceTest {
```

Comment:
`DaggerNestedDependencyInterfaceTest` is generated by traditional dagger, which is only executed on production code:
https://github.com/JakeWharton/dagger-reflect/blob/b009ca8d3cdd6dac0a6c653ea4459e02996a540e/integration-tests/build.gradle#L8

Move this class to `src/main/java`.